app/main.py

```diff
@@ -1,2 +1,114 @@
+from dataclasses import dataclass
+from typing import Any, Hashable, List, Optional, Iterator
+
+
+INITIAL_CAPACITY = 8
+LOAD_FACTOR = 0.75
+
+
+@dataclass
+class Node:
+    key: Hashable
+    value: Any
+    _hash: int
+
+
 class Dictionary:
-    pass
+    def __init__(self, capacity: int = INITIAL_CAPACITY) -> None:
+        self.capacity = capacity
+        self.size = 0
+        self.hash_table: List[Optional[Node]] = [None] * self.capacity
+
+    def calculate_index(self, key: Hashable, key_hash: int) -> int:
+        index = key_hash % self.capacity
+        while (
+                self.hash_table[index] is not None
+                and self.hash_table[index].key != key
+        ):
+            index = (index + 1) % self.capacity
+        return index
+
+    def resize(self) -> None:
+        old_hash_table = self.hash_table
+        self.capacity *= 2
+        self.hash_table = [None] * self.capacity
+        self.size = 0
+
+        for node in old_hash_table:
+            if node is not None:
+                self.__setitem__(node.key, node.value)
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        key_hash = hash(key)
+        index = self.calculate_index(key, key_hash)
+
+        if self.hash_table[index] is None:
+            if self.size + 1 > self.capacity * LOAD_FACTOR:
+                self.resize()
+                return self.__setitem__(key, value)
```

Comment:
When resizing the hash table, the method should not return immediately after calling `resize()`. Instead, it should continue to insert the current key-value pair into the resized hash table. The current implementation will skip adding the item that triggered the resize.

Reply:
at __setitem__ you don't need to return smth. 
If you resize you simply go out (return)